### ISSUE TITLE
[Bug]:  When the table with deletefile in V2 format plan, the phenomenon of Icebergdeletefile amplification will appear

### ISSUE THREAD

**Issue by huyuanfeng2018** [NetEase/arctic#1730]
### What happened?

<img width="1029" alt="image" src="https://github.com/NetEase/arctic/assets/40817998/8d6aeb82-5c92-4ea6-83ac-7592c0ae3808">
IcebergDeleteFile wraps GenericDeleteFile, but because there are duplicate deletefile objects in FileScanTask, GenericDeleteFile will be wrapped repeatedly, resulting in the explosion of the number of IcebergDeleteFile objects
<img width="582" alt="image" src="https://github.com/NetEase/arctic/assets/40817998/771bcc79-d113-4896-b39a-4297370cf9d3">
<img width="721" alt="image" src="https://github.com/NetEase/arctic/assets/40817998/045169ea-a616-435c-b5d5-44ebf7bbad0f">

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

in any case

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by wangtaohz** [comment]
Thanks for your report!

I think deduplicating delete files based on their paths and avoiding creating too many IcebergDeleteFile objects can solve the problem.
```
private IcebergDeleteFile createDeleteFile(DeleteFile deleteFile) {
    return new IcebergDeleteFile(deleteFile, sequenceNumberFetcher.sequenceNumberOf(deleteFile.path().toString()));
  }
```

We should fix it ASAP. Would you be willing to submit a PR to fix this issue? @huyuanfeng2018

**Comment by huyuanfeng2018** [comment]
> Thanks for your report!
> 
> I think deduplicating delete files based on their paths and avoiding creating too many IcebergDeleteFile objects can solve the problem.
> 
> ```
> private IcebergDeleteFile createDeleteFile(DeleteFile deleteFile) {
>     return new IcebergDeleteFile(deleteFile, sequenceNumberFetcher.sequenceNumberOf(deleteFile.path().toString()));
>   }
> ```
> 
> We should fix it ASAP. Would you be willing to submit a PR to fix this issue? @huyuanfeng2018

This has been fixed in #1731
